src/Components/Login.js

```diff
@@ -1,53 +1,104 @@
+import LockOutlinedIcon from "@material-ui/icons/LockOutlined"
+import Typography from "@material-ui/core/Typography"
+import { makeStyles } from "@material-ui/core/styles"
+import Container from "@material-ui/core/Container"
+import { Alert } from "@material-ui/lab"
+import { Redirect } from "react-router"
 
-import React from 'react';
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(8),
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.primary.dark,
+  },
+  form: {
+    width: "100%",
+    marginTop: theme.spacing(1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}))
 
-import {Redirect} from 'react-router-dom'
-const Login = ({loginhandler,inputName,inputpassword,nameChangeHandler,passwordChangeHandler,redirect}) => {
+export default function Login({
+  loginhandler,
+  inputName,
+  inputpassword,
+  nameChangeHandler,
+  passwordChangeHandler,
+  redirect,
+  alert,
+}) {
+  const classes = useStyles()
 
-   
-    return (
-      <div>
-        {redirect ? (
-          <Redirect to="/todo" />
-        ) : (
-          <div className="container h-100">
-            <div className="row h-100 justify-content-center align-items-center">
-              <h2 className="header">Sign in to your account</h2>
-              <div className="form-container col-md-6">
-                <form className="login-form" onSubmit={loginhandler}>
-                  <div>
-                    <label htmlFor="name">Name</label>
-                    <input
-                      className="login-input"
-                      value={inputName}
-                      name="name"
-                      type="text"
-                      onChange={nameChangeHandler}
-                    />
-                  </div>
-
-                  <div>
-                    <label htmlFor="password">Password</label>
-
-                    <input
-                      value={inputpassword}
-                      name="password"
-                      type="password"
-                      onChange={passwordChangeHandler}
-                    />
-                  </div>
-                  <div className="text-center">
-                    <button className="submit-btn" type="submit">
-                      Login
-                    </button>
-                  </div>
-                </form>
-              </div>
-            </div>
+  return (
+    <>
+      {redirect ? (
+        <Redirect to="/todo" />
+      ) : (
+        <Container maxWidth="xs">
+          <CssBaseline />
+          <div className={classes.paper}>
+            <Avatar className={classes.avatar}>
+              <LockOutlinedIcon />
+            </Avatar>
+            <Typography component="h1" variant="h5">
+              Sign in
+            </Typography>
+            <form className={classes.form} noValidate onSubmit={loginhandler}>
```

Comment:
Can we call function onclick in button rather than onSubmit?

Reply:
I have used e.preventdefault to prevent refresh on submit. So it doesn't matter ig. If it does then I can change it

Reply:
OnClick wont work on enter while onSubmit on form handles on enter as well